### PR TITLE
Feature/simple react portlet

### PR DESCRIPTION
**Pull request by raptor22fa**
@tommmyy Jirka will do the code review.

### CODE REVIEW

File: boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition;
+import eu.reactunion.boilerplate.constants.HeroPortletKeys;
+
+/**
+ * Configuration class of the Hero portlet.
+ *
+ * @author Roman Srom (roman.srom@lundegaard.eu)
+ */
+@ExtendedObjectClassDefinition(
+        category = "ReactUnion",
```
Comment (Juldik) [inline boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfiguration.java]: I would prefer to use one of the predifined categories than creating new for react uinon (e.g. Foundation).

File: boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfigurationUtil.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    /**
+     * Adds configuration properties to a render request as the attributes.
+     *
+     * @param request render request
+     * @throws ConfigurationException if the portlet configuration can't be retrieved
+     */
+    public static void addConfigurationContext(RenderRequest request) throws ConfigurationException {
+        final ThemeDisplay themeDisplay = (ThemeDisplay) request.getAttribute(WebKeys.THEME_DISPLAY);
+        if (themeDisplay != null) {
+            final PortletDisplay portletDisplay = themeDisplay.getPortletDisplay();
```
Comment (Juldik) [inline boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfigurationUtil.java]: I would consider creating getHeroConfiguration(ThemeDisplay themeDisplay) as a private method. Both method uses similar code.

File: boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfigurationAction.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    service = ConfigurationAction.class
+)
+public class HeroConfigurationAction extends DefaultConfigurationAction {
+
+    @Reference
+    private ConfigurationProvider configurationProvider;
+
+    @Override
+    public void processAction(PortletConfig portletConfig, ActionRequest actionRequest, ActionResponse actionResponse) throws Exception {
+        String heading = ParamUtil.getString(actionRequest, HeroConstants.PARAM_HEADING);
```
Comment (Juldik) [inline boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfigurationAction.java]: If no validation required it is possible to use preferences naming convetion (--pereference--heading) in aui input name and storing of preferences is done automatically.

File: boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        scope = ExtendedObjectClassDefinition.Scope.PORTLET_INSTANCE
+)
+@Meta.OCD(id = HeroPortletKeys.CONFIGURATION)
+public interface HeroConfiguration {
+
+    // required true doesn't work, it doesn't find default values. Probably bug.
+    @Meta.AD(
+            required = false
```
Comment (Juldik) [inline boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfiguration.java]: for default values try using deflt = "value". This usually works well.

File: boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfiguration.java (hunk 1 above)
Comment (raptor22fa) [reply boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfiguration.java]: Implemented

File: boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfigurationUtil.java (hunk 2 above)
Comment (raptor22fa) [reply boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfigurationUtil.java]: Implemented

File: boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfigurationAction.java (hunk 3 above)
Comment (raptor22fa) [reply boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfigurationAction.java]: Implemented

**Pull request merged**